tools/ockam-hub-cli/src/auth/github.rs

```diff
@@ -0,0 +1,119 @@
+use colored::*;
+use std::collections::HashMap;
+use std::future::Future;
+use std::pin::Pin;
+use webbrowser;
+
+const GITHUB_CLIENT_ID: &'static str = "609323e5ba23958cb2f5";
+
+pub async fn post(
+    client: &reqwest::Client,
+    url: &str,
+    payload: &HashMap<&str, &str>,
+) -> Result<HashMap<String, String>, reqwest::Error> {
+    let response = client.post(url).json(&payload).send().await?.text().await?;
+
+    let parsed_response =
+        url::form_urlencoded::parse(response.as_bytes()).fold(HashMap::new(), |mut acc, s| {
+            acc.insert(s.0.to_string(), s.1.to_string());
+            acc
+        });
+
+    Ok(parsed_response)
+}
+
+pub fn try_access_token<'a>(
+    client: &'a reqwest::Client,
+    payload: &'a HashMap<&str, &str>,
+    time_to_retry: u64,
+) -> Pin<Box<dyn Future<Output = Result<HashMap<String, String>, reqwest::Error>> + 'a>> {
+    Box::pin(async move {
+        let response = post(
+            client,
+            "https://github.com/login/oauth/access_token",
+            payload,
+        )
+        .await?;
+
+        if let Some(error) = response.get("error") {
+            if error == "authorization_pending" {
+                let duration = std::time::Duration::from_secs(time_to_retry);
+                std::thread::sleep(duration);
+                return try_access_token(client, payload, time_to_retry).await;
+            }
```

Comment:
Can we add an `else` here that will panic on any other kind of error?

Reply:
Sry for the long time to fix, I was busy this week :/